### PR TITLE
- Changes for Ideas routes and data: Get-Ideas, Update-Idea, Delete-I…

### PR DESCRIPTION
**Pull request by 1994nikunj**
- Almost all Ideas backend changes
- Changes for better code readability

### CODE REVIEW

File: server/data/ideas.js [hunk 1]
```diff
@@ -1,111 +1,286 @@
...
 const { isValidUsername } = require('../utils/users');
 const { isValidTechnologies } = require('../utils/projects');
-const {
-	internalServerErr,
-	badRequestErr,
-	notFoundErr,
-} = require('../utils/index');
+const { internalServerErr, badRequestErr, notFoundErr } = require('../utils/index');
 
+const helper = require('../utils/ideas');
```
Comment (GiriNithin) [inline server/data/ideas.js:10]: @1994nikunj Why are you importing the entire helper file.. just destructure the required function and use it.

File: server/data/ideas.js [hunk 2]
```diff
@@ -1,111 +1,286 @@
...
-const { isValidLookingFor, isValidStatus } = require('../utils/ideas');
 const { isValidProjectName } = require('../utils/projects');
 const { isValidUsername } = require('../utils/users');
 const { isValidTechnologies } = require('../utils/projects');
-const {
-	internalServerErr,
-	badRequestErr,
-	notFoundErr,
-} = require('../utils/index');
+const { internalServerErr, badRequestErr, notFoundErr } = require('../utils/index');
```
Comment (GiriNithin) [inline server/data/ideas.js:8]: Are you not using prettier and eslint ??. We are all following eslint and prettier indentation. Make sure you follow the same.

File: server/data/ideas.js [hunk 3]
```diff
@@ -1,111 +1,286 @@
...
+	if (!createIdeaAck?.acknowledged || !createIdeaAck?.insertedId) {
 		throw internalServerErr('Could not create idea. Please try again');
+	}
+
 	const createdIdea = await getIdeaById(
-		createIdeaAcknowledgement.insertedId.toString()
+		createIdeaAck.insertedId.toString()
 	);
+
 	return createdIdea;
```
Comment (GiriNithin) [inline server/data/ideas.js:69]: Bro.. what is the point of making createIdeaAcknowledgment into createIdeaAck ??

File: server/data/ideas.js [hunk 4]
```diff
@@ -1,111 +1,286 @@
...
 			{ _id: ObjectId(likeIdeaId) },
 			{ $set: { likes: getIdeaInfo.likes } }
 		);
-		if (
-			!likeIdeaAcknowledgment.acknowledged ||
-			!likeIdeaAcknowledgment.modifiedCount
-		)
+		if (!likeIdeaAck.acknowledged || !likeIdeaAck.modifiedCount) {
 			throw internalServerErr('Could not like the idea. Please try again.');
-	} else throw badRequestErr('Idea already liked.');
```
Comment (GiriNithin) [inline server/data/ideas.js:79]: Same here you are just changing variable names

File: server/data/ideas.js [hunk 5]
```diff
@@ -1,111 +1,286 @@
...
 			{ _id: ObjectId(ideaId) },
 			{ $pull: { likes: ObjectId(userId) } }
 		);
-		if (
-			!unlikeIdeaAcknowledgment.acknowledged ||
-			!unlikeIdeaAcknowledgment.modifiedCount
-		)
+		if (!unlikeIdeaAck.acknowledged || !unlikeIdeaAck.modifiedCount) {
 			throw internalServerErr('Could not unlike the idea. Please try again.');
-	} else throw badRequestErr('Idea already unliked.');
```
Comment (GiriNithin) [inline server/data/ideas.js:101]: Same here

File: server/routes/ideas.js [hunk 6]
```diff
@@ -1,81 +1,268 @@
...
+
 const { successStatusCodes, badRequestErr } = require('../utils');
 const { authenticateToken } = require('../middleware/auth');
-const {
-	isValidIdeaName,
-	isValidStatus,
-	isValidLookingFor,
-} = require('../utils/ideas');
+const { isValidIdeaName, isValidStatus, isValidLookingFor } = require('../utils/ideas');
 const { isValidTechnologies } = require('../utils/projects');
```
Comment (GiriNithin) [inline server/routes/ideas.js:10]: Again make sure you are using eslint and prettier

File: server/routes/ideas.js [hunk 7]
```diff
@@ -1,81 +1,268 @@
...
+
+
+// Like / Dislike an Idea
+// ======================================================================
 router
 	.route('/:ideaId/likes')
-	.post(authenticateToken, async (req, res) => {
-		const { user } = req;
+	.post(authenticateToken, async (request, response) => {
+		const { user } = request;
```
Comment (GiriNithin) [inline server/routes/ideas.js:159]: We are follow req, res as the naming convention

File: server/routes/ideas.js [hunk 8]
```diff
@@ -1,81 +1,268 @@
...
-				unlikeIdeaInfo,
-			});
+
+			response
+				.status(successStatusCodes.DELETED)
+				.json({
+					unlikeIdeaInfo,
+				});
+
 		} catch (e) {
```
Comment (GiriNithin) [inline server/routes/ideas.js:196]: Same here because of that change in variable name.

File: server/routes/ideas.js [hunk 9]
```diff
@@ -1,81 +1,268 @@
...
+
+			response
+				.status(successStatusCodes.CREATED)
+				.json({
+					comments
+				});
+
+		} catch (e) {
+			sendErrResp(response, e);
+		}
```
Comment (GiriNithin) [inline server/routes/ideas.js:234]: In createIdea you just added comments for each and every single validations and for createComment you did not.. There is no consistency if possible try to remove comments in every file everybody knows what is happening in the code and what they did.

File: server/routes/ideas.js [hunk 10]
```diff
@@ -1,81 +1,268 @@
...
+				.status(successStatusCodes.CREATED)
+				.json({
+					idea
+				});
+
+		} catch (e) {
+			sendErrResp(response, e);
+		}
+	});
+
```
Comment (GiriNithin) [inline server/routes/ideas.js:64]: Just for the sake of adding comments you could have just changed where it was needed.

**Review by GiriNithin — CHANGES_REQUESTED** [review]
Make sure you are using eslint and prettier. Don't change variable names for the sake of changing and also remove comments in all files not required for now.. We can do that 1-2 days before the submission if required. Also, rather than importing entire helpers try to destructure what is required by you.

**Review by khatri7 — APPROVED** [review]

**Pull request merged**